### PR TITLE
Rate Card Changes

### PR DESCRIPTION
**Pull request by corymcdonald**
## Rate Card Changes

localhost_3000_publishers_home_locale=en (2)

#### Features

- Gives UI indications for number of confirmations, installs, downloads per month
- Shows to user total amount in the referral charts
- Moves the edit contact form to the settings page

#### How To Test

Add a channel
Activate the referral promo
Run  `rails database_updates:mock_data:populate_promo_stats` to update the data for the server
Navigate around the website and test different behaviors

### CODE REVIEW

File: app/views/layouts/promo_registrations.html.slim [hunk 1]
```diff
@@ -3,15 +3,16 @@
 
 - content_for(:nav) do
   .nav.nav--promo
-    = link_to(root_path) do
-      .brave-logo
-        = render "logo_wordmark_white_svg"
+    / = link_to(root_path) do
+    /   .brave-logo
```
Comment (yachtcaptain23) [inline app/views/layouts/promo_registrations.html.slim]: Can we just remove these if we're not going to use them?

Comment (corymcdonald) [reply app/views/layouts/promo_registrations.html.slim]: https://github.com/brave-intl/publishers/pull/2333/commits/86c918bf002a417ed15a74c395eb8246a7429b96

File: app/services/eyeshade/referrals.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+class Eyeshade::Referrals < Eyeshade::BaseApiClient
+  PATH = "/v1/referrals"
+
+  def groups
+    return offline_groups if Rails.application.secrets[:api_eyeshade_offline]
+
+    response = connection.get do |request|
```
Comment (yachtcaptain23) [inline app/services/eyeshade/referrals.rb]: I think this should be RESTful

File: app/services/eyeshade/referrals.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    JSON.parse(response.body)
+  end
+
+  def statement(publisher:, start_date:, end_date:)
+    return offline_statement if Rails.application.secrets[:api_eyeshade_offline]
+
+    response = connection.get do |request|
+      request.headers["Authorization"] = api_authorization_header
+      request.url(PATH + "/statement/#{publisher}")
```
Comment (yachtcaptain23) [inline app/services/eyeshade/referrals.rb]:
```suggestion
      request.url(PATH + "/statement/#{publisher.id}")
```

File: app/models/promo_registration.rb [hunk 4]
```diff
@@ -102,16 +104,18 @@ def stats_by_date
...
-        FIRST_RUNS => 0,
-        FINALIZED => 0,
-      }) do |aggregate_stats, promo_registration|
-        promo_registration_aggregate_stats = promo_registration.aggregate_stats
-        aggregate_stats[RETRIEVALS] += promo_registration_aggregate_stats[RETRIEVALS]
-        aggregate_stats[FIRST_RUNS] += promo_registration_aggregate_stats[FIRST_RUNS]
-        aggregate_stats[FINALIZED] += promo_registration_aggregate_stats[FINALIZED]
+    def stats_for_registrations(promo_registrations:, start_date: Date.new, end_date: Date.today.at_end_of_day)
+      promo_registrations.
+        map { |p| p.aggregate_stats_by_period(start_date, end_date) }.
```
Comment (yachtcaptain23) [inline app/models/promo_registration.rb]:
```suggestion
        map { |pr| pr.aggregate_stats_by_period(start_date, end_date) }.
```

**Pull request merged**